### PR TITLE
Replaced deprecated IE API

### PR DESCRIPTION
**Pull request by ivikhrev**
This PR replaces API that was deprecated in the latest IE version and fixes next warnings:
```
warning C4996: 'InferenceEngine::IInferRequest': Do not use IInferRequest API
 
warning C4996: 'InferenceEngine::make_so_pointer': Use std::make_shared<Extension>
 
warning C4996: 'InferenceEngine::ExecutableNetwork::CreateInferRequestPtr': Use ExecutableNetwork::CreateInferRequest instead
```

### CODE REVIEW

File: demos/smart_classroom_demo/cpp/src/detector.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ void FaceDetection::submitRequest() {
 
 void FaceDetection::enqueue(const cv::Mat &frame) {
     if (!request) {
-        request = net_.CreateInferRequestPtr();
+        request = std::make_shared<InferenceEngine::InferRequest>(net_.CreateInferRequest());
```
Comment (Wovchena) [inline demos/smart_classroom_demo/cpp/src/detector.cpp:60]: Is it possible to replace a ptr to `InferRequest` with actual `InferRequest` in such cases?

**Review by IRDonch — APPROVED** [review]
I agree with @Wovchena's comment about removing `InferRequest::Ptr`, but I'd like to finish the upgrade to new IE ASAP (and this PR is a prerequisite), so I'm going to leave that for a potential future PR.

**Comment by ivikhrev** [comment]
@IRDonch @Wovchena ok. I will try to do such fixes.

File: demos/interactive_face_detection_demo/cpp/detectors.cpp [hunk 2]
```diff
@@ -693,7 +693,7 @@ void FacialLandmarksDetection::enqueue(const cv::Mat &face) {
         return;
     }
     if (!request) {
-        request = net.CreateInferRequestPtr();
+        request = std::make_shared<InferenceEngine::InferRequest>(net.CreateInferRequest());
```
Comment (ilya-lavrenov) [inline demos/interactive_face_detection_demo/cpp/detectors.cpp:696]: BTW, do you need shared_ptr? InferRequest itself is a wrapper for shared_ptr<InternalInferRequestInterface>

Comment (ilya-lavrenov) [reply demos/interactive_face_detection_demo/cpp/detectors.cpp:696]: ok, I see you are going to do this separately.

**Pull request merged**